src/jumps.jl

```diff
@@ -81,14 +110,29 @@ struct VariableRateJump{R, F, I, T, T2} <: AbstractJump
 end
 
 function VariableRateJump(rate, affect!;
+                          lrate = nothing, urate = nothing,
+                          L = nothing, rootfind = true,
                           idxs = nothing,
-                          rootfind = true,
-                          save_positions = (true, true),
+                          save_positions = (lrate !== nothing && urate !== nothing &&
+                                            L !== nothing) ? (false, true) : (true, true),
                           interp_points = 10,
                           abstol = 1e-12, reltol = 0)
-    VariableRateJump(rate, affect!, idxs,
-                     rootfind, interp_points,
-                     save_positions, abstol, reltol)
+    if !((lrate === nothing && urate === nothing && L === nothing) ||
+         (lrate !== nothing && urate !== nothing && L !== nothing))
+        error("Either `lrate`, `urate` and `L` must be nothing, or all of them must be defined.")
+    end
+
+    VariableRateJump(rate, affect!, lrate, urate, L, idxs, rootfind,
+                     interp_points, save_positions, abstol, reltol)
+end
+
+function VariableRateJump(jump::ConstantRateJump)
```

Comment:
The problem is that the aggregators are not jump "aware", which also came as a surprise to me. When `JumpProblem` is initialized it calls `aggregate` to initialize the aggregator. This function expects an iterable of jumps which does not distinguish between `ConstantRateJump` and `VariableRateJump`. 

I defined this constructor to be used internally, it makes it easier to construct an iterable of `VariableRateJump` to pass to the `aggregate` function of `Coevolve`. As such, I do not mention this in the documentation as an option for creating `VariableRateJump`s.

As you suggested below, we could use the index of the jump to distinguish between `MassActionJump`, `ConstantRateJump` and `VariableRateJump`. Problems arise when the aggregator expects a dependency graph. In this case, the user needs to be aware of this rule so that when he passes `VariableRateJump`s before `ConstantRateJump`s. It must be clear whether graph indices should be declared in the order passed by the user or in the order that the aggregator expects. An option would be to take the dependency graph passed by the user and re-map it internally to the way we sort the indices for the aggregator. However, I feel this might be prone to errors and more difficult to debug for the final user since graph indices in the aggregator will not correspond to the one used to initialize the problem.

Alternatively, the aggregator could keep a vector that tracks the type of jump for each index. We could pass this vector when calling `aggregate`. This might exert a penalty cost, but might be cleaner and leave the door open for other algorithms to distinguish between jump types in any order.

Reply:
Can't you just add a kwarg to hold the `VariableRateJump`s in `aggregate`? 

Let's just assume we order `MassActionJump`s, `ConstantRateJump`s and then `VariableRateJump`s in dependency graphs. We already assume the first two have that order now (and say so in the docs -- but perhaps this needs to be made more clear).

Then you can assume `rates` are ordered first by `ConstantRateJump`s and then by `VariableRateJump`s, or just make a separate `vrjrates` field. Likewise you can shorten `urates`, `lrates` and `Ls` to just have length equal to the number of `VariableRateJump`s.

Reply:
I have removed the `VariableRateJump` initializer from a `ConstantRateJump`. 

Now, I understand your suggestion. `JumpSet` will split the jumps into the three categories, so their indices will invariably need to be ordered in blocks.

I have bundled `ConstantRateJump`s with `VariableRateJump`s in a vector. Then passing this vector to `aggregate`. I have not created a separate field `vrjrates` as it is not needed.